### PR TITLE
feat(general): Added computation of git_root_path to igraph serialization

### PR DESCRIPTION
**Pull request by bo156**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Save the root path for the scanned repo in order to do comparsions on relative paths if needed.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nimrodkor — APPROVED** [review]

File: checkov/common/util/igraph_serialization.py [hunk 1]
```diff
@@ -7,7 +8,7 @@
     from igraph import Graph
 
 
-def serialize_to_json(igraph: Graph) -> Dict[str, Any]:
+def serialize_to_json(igraph: Graph, absolute_root_folder: str) -> Dict[str, Any]:
```
Comment (YaaraVerner) [inline checkov/common/util/igraph_serialization.py:11]: can you make the new parameter optional? we copied the serializer here and will need to remove it from the platform and use this function instead. In the other cases, we won't save the `absolute_root_folder`

File: checkov/common/util/igraph_serialization.py [hunk 2]
```diff
@@ -26,5 +27,6 @@ def serialize_to_json(igraph: Graph) -> Dict[str, Any]:
         for edge in igraph.es
     ]
 
-    graph = {"graph_type": "igraph", "nodes": nodes, "links": links}
+    graph = {"graph_type": "igraph", "git_root_path": absolute_root_folder, "nodes": nodes, "links": links}
```
Comment (YaaraVerner) [inline checkov/common/util/igraph_serialization.py:30]:
```suggestion
    graph = {"graph_type": "igraph", "absolute_root_folder": absolute_root_folder, "nodes": nodes, "links": links}
```

**Review by YaaraVerner — APPROVED** [review]
Nice work!

File: checkov/common/util/igraph_serialization.py (hunk 1 above)
Comment (bo156) [reply checkov/common/util/igraph_serialization.py]: fixed

**Pull request merged**